go.mod

```diff
@@ -121,3 +121,5 @@ require (
 	sigs.k8s.io/json v0.0.0-20241010143419-9aa6b5e7a4b3 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.2 // indirect
 )
+
+replace sigs.k8s.io/karpenter => github.com/jigisha620/karpenter v0.0.0-20250203191516-1a223f0e374d
```

Comment:
Will revert this once the upstream PR is merged - https://github.com/kubernetes-sigs/karpenter/pull/1944